internal/collector/postgres_metrics.go

```diff
@@ -75,6 +78,13 @@ func EnablePostgresMetrics(ctx context.Context, inCluster *v1beta1.PostgresClust
 			log.Error(err, "error compiling postgres metrics")
 		}
 
+		if inCluster.Spec.PostgresVersion == 16 {
+			fiveSecondMetricsClone, err = appendToJSONArray(fiveSecondMetricsClone, eqPG16)
+		}
+		if err != nil {
+			log.Error(err, "error compiling postgres metrics")
```

Comment:
Now that I look at this again, I wish the errors had some differentiation so that we could pinpoint which `appendToJSONArray` actually errored. (This may be scope creep as it doesn't entirely fall within this ticket.)

Reply:
I'd like to get this merged sooner than later so I can move on to fixing the intervals for certain queries; however, I'm going to be touching this code again when working on that, so I could improve the error messages then